modules/authentication/components/ForgotPassword.jsx

```diff
@@ -0,0 +1,66 @@
+import React, {
+  cloneElement,
+  Component,
+  PropTypes
+} from 'react';
+
+const propTypes = {
+  actions: PropTypes.shape({
+    authentication: PropTypes.shape({
+      requestPasswordReset: PropTypes.func.isRequired
+    }).isRequired
+  }).isRequired,
+  children: PropTypes.node,
+  location: PropTypes.shape({
+    query: PropTypes.shape({
+      next: PropTypes.string
+    }).isRequired
+  }).isRequired,
+  router: PropTypes.shape({
+    replace: PropTypes.func.isRequired
+  })
+};
+
+class ForgotPassword extends Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    const values = Array.prototype.reduce.call(this.forgotPasswordForm.elements, (memo, element) => {
+      if (element.name && element.value) {
+        memo[element.name] = element.value;
+      }
+
+      return memo;
+    }, {});
```

Comment:
I guess you could move it to a utility function. But make it a single file per exported function.

so in `modules/authentication/utils/mapFormValues.js`:
```
function mapFormValues(formElements) {
  return Array.prototype.reduce.call(formElements, (memo, element) => {
    if (element.name && element.value) {
      memo[element.name] = element.value;
    }

    return memo;
  }, {});
}
export default mapFormValues
```

no more giant files with a bunch of unrelated functions, they are garbage heaps of waste